0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+= LND as Alpha Ledger as described in RFC003
+
+NOTE: Authors: Philipp Hoenisch +
+Date: 2019-04-24 +
+Tracking issue: TBA
+
+:toc:
+
+== Context
+
+The **Hold invoice feature** was introduced in the following PR: https://github.com/lightningnetwork/lnd/pull/2022[lightningnetwork/lnd/pull/2022].
+____
+A hold invoice is a new type of invoice that triggers a different flow at the receiver end.
+Instead of immediately locking in and settling the htlc when the payment arrives, the htlc for a hold invoice is only locked in and not yet settled.
+At that point, it is not possible anymore for the sender to revoke the payment, but the receiver still can choose whether to settle or cancel the htlc and invoice.
+____
+
+This allows us to trade Bitcoin on the Lightning Network (LN) for an asset _X_ on a different ledger where LN is the _alpha ledger_ as per the definition in https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md[RFC003].
+For the other way around, Bitcoin on the Lightning Network is the _beta asset/ledger_, we can make use of `createinvoice` as Alice would instantly release the secret by settling Bob's LN payment.
+
+=== Standard _hold invoice_ process
+
+____
+Assumption: Alice wants to pay Bob _X_ Bitcoin using the LN but Bob does not generate the preimage - Alice does it.
+____
+
+1) Alice generates a secret preimage `p` and sends the hash of it `h(p)` to Bob.
+2) Bob calls `addholdinvoice` with an amount `X` and the hashed preimage `h(p)` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[more details]).
+3) Bob sends the generated invoice to Alice.
```

Comment:
Is this necessary? Alice should be able to generate that invoice by herself right?

Reply:
Yes @thomaseizinger is correct.

Reply:
True that, it's possible. Two points come to my mind: 
 * By doing this we identify an invoice by the `secret_hash` and not by the invoice itself ala [bolt 11](https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md). Meaning, Bob will need to watch for an invoice by the `secret_hash` itself -> which is fine. 
* We cannot make use of features such as a signed invoice, optional proposed routing table, description, etc, -> which might be useful in the future

Reply:
> PS what's a signed invoice?
To verify that you are not paying an invoice of someone else?